docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/rs-configuration.md

```diff
@@ -182,16 +182,259 @@ The capacity usage on a pool can be monitored using [exporter metrics](../replic
+
+**Command**
+```text
+cat <<EOF | kubectl create -f -
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: mayastor-1
+parameters:
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "2"
+  nodeSpreadTopologyKey: |
+    zone
+provisioner: io.openebs.csi-mayastor
+volumeBindingMode: Immediate
+```
+
+Apply the labels to the nodes using the below command:
+
+**Command**
+```text
+kubectl mayastor label node worker-node-1 zone=us-west-1
+kubectl mayastor label node worker-node-2 zone=eu-east-1
+kubectl mayastor label node worker-node-3 zone=us-north-1
+ ```
+
+**Command (Get nodes)**
+ ```text
+kubectl mayastor get nodes -n openebs --show-labels
+ID             GRPC ENDPOINT        STATUS  LABELS
+worker-node-1  65.108.91.181:10124  Online  zone=eu-west-1
+worker-node-3  65.21.4.103:10124    Online  zone=eu-east-1
+worker-node-3  37.27.13.10:10124    Online  zone=us-west-1
+```
+
+In this case, the replica of the volume will be placed on any nodes i.e.
+- `worker-node-1` and `worker-node-2` or
+- `worker-node-2` and `worker-node-3`
+as the storage class has `zone` as the value for `nodeSpreadTopologyKey` that matches the label key of the node but has a different value.
+
+### "poolAffinityTopologyLabel"
+
+The parameter 'poolAffinityTopologyLabel' will allow the placement of replicas on the pool that exactly match the labels defined in the storage class.
+
+For the case shown below, the replica of the volume will be placed on `pool-on-node-0` and `pool-on-node-3` only as they match the labels specified under `poolAffinityTopologyLabel` in the storage class that is equal to zone=us-west-1.
+
+**Command**
+```text
+cat <<EOF | kubectl create -f -
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: mayastor-1
+parameters:
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "2"
+  poolAffinityTopologyLabel: |
+    zone: us-west-1
+provisioner: io.openebs.csi-mayastor
+volumeBindingMode: Immediate
+```
+
+Apply the labels to the nodes using the below command:
+
+**Command**
+```text
+cat <<EOF | kubectl create -f -
+apiVersion: "openebs.io/v1beta2"
+kind: DiskPool
+metadata:
+  name: pool-on-node-0
+  namespace: mayastor
+spec:
+  node: worker-node-0
+  disks: ["/dev/sdb"]
+  topology:
+    labelled:
+        zone: us-west-1
+---
+apiVersion: "openebs.io/v1beta2"
+kind: DiskPool
+metadata:
+  name: pool-on-node-1
+  namespace: mayastor
+spec:
+  node: worker-node-1
+  disks: ["/dev/sdb"]
+  topology:
+    labelled:
+        zone: us-east-1
+---
+apiVersion: "openebs.io/v1beta2"
+kind: DiskPool
+metadata:
+  name: pool-on-node-2
+  namespace: mayastor
+spec:
+  node: worker-node-2
+  disks: ["/dev/sdb"]
+  topology:
+    labelled:
+        zone: us-west-1
+ ```
+
+ **Command (Get filtered pools based on labels)**
+ ```text
+kubectl mayastor get pools -n openebs --selector zone=eu-west-1
```

Comment:
We should use the 'kubectl get diskpool' command ideally IMO. Just my 2c.

Reply:
Agreed, but we can take this later, for now the the thing which is implemented is documented above.